testnet/stacks-node/src/tests/signer.rs

```diff
@@ -329,3 +333,105 @@ fn test_stackerdb_dkg() {
+    let signer_configs = build_signer_config_tomls(
+        &signer_stacks_private_keys,
+        num_keys,
+        &conf.node.rpc_bind,
+        &stacker_db_contract_id.to_string(),
+        Some(&pox_contract_id.to_string()),
+        Some(Duration::from_millis(128)), // Timeout defaults to 5 seconds. Let's override it to 128 milliseconds.
+    );
+
+    let mut running_signers = vec![];
+    // Spawn all the signers first to listen to the coordinator request for dkg
+    let mut signer_cmd_senders = Vec::new();
+    let mut signer_res_receivers = Vec::new();
+    let mut ping_rxs = vec![];
+    for i in (0..num_signers).rev() {
+        let (cmd_send, cmd_recv) = channel();
+        let (res_send, res_recv) = channel();
+        let (running_signer, ping_recv) =
+            spawn_signer(&signer_configs[i as usize], cmd_recv, res_send);
+        ping_rxs.push(ping_recv);
+        running_signers.push(running_signer);
+        cmd_send
+            .send(RunLoopCommand::Ping { payload_size: 0 })
+            .unwrap();
+        signer_cmd_senders.push(cmd_send);
+        signer_res_receivers.push(res_recv);
+    }
+
+    let coordinator_cmd_send = signer_cmd_senders.last().unwrap().clone();
+    let coordinator_res_recv = signer_res_receivers.last().unwrap();
+
+    // Let's scope the node to ensure stopping the signers doesn't cause issues.
+    {
+        // Build our simulated pox-4 stacks contract TODO: replace this with the real deal?
+        let pox_contract = build_pox_contract(num_signers);
+        let signer_stacks_addresses = signer_stacks_private_keys
+            .iter()
+            .map(to_addr)
+            .collect::<Vec<StacksAddress>>();
+        // Build the stackerdb contract
+        let stackerdb_contract = build_stackerdb_contract(&signer_stacks_addresses);
+
+        // Setup the nodes and deploy the contract to it
+        let _node = setup_stx_btc_node(
+            &mut conf,
+            num_signers,
+            &signer_stacks_private_keys,
+            &publisher_private_key,
+            &stackerdb_contract,
+            &stacker_db_contract_id,
+            &pox_contract,
+            &pox_contract_id,
+            &signer_configs,
+        );
+
+        coordinator_cmd_send
+            .send(RunLoopCommand::Sign {
+                message: vec![1, 2, 3, 4, 5],
+                is_taproot: false,
+                merkle_root: None,
+            })
+            .expect("failed to send Sign command");
+
+        'outer: loop {
+            let results = coordinator_res_recv.recv().expect("failed to recv results");
+            for result in results {
+                if let OperationResult::Sign(_) = result {
+                    break 'outer;
+                };
+            }
+        }
+    }
+
+    for signer in running_signers {
+        assert!(signer.stop().is_none());
+    }
+
+    for rx in ping_rxs {
+        rx.recv().unwrap();
+        assert_eq!(rx.recv(), Err(RecvError));
```

Comment:
I'm not seeing in this test where the ping/pong happens?

Reply:
[Here](https://github.com/stacks-network/stacks-core/pull/4167/files#diff-a2df5fcad27701f7a726d41b6fa85dfaa3382377e1a1fbb6f03e58f11047dbb3R378) a Ping command is sent to each signer's runloop. This means there will be a single RTT per signer in the Receiver<(u64,Duration)> end. We assert that there is exactly one RTT per signer as expected. The Receiver would normally be consumed upon instantiating a PingerThread, but that is covered already in another test.

Reply:
Thanks! :pray: